src/services/agile/providers/azureDevOps/azureDevOpsUtils.ts

```diff
@@ -0,0 +1,103 @@
+import { BacklogItem, BacklogItemType } from "../../../../models";
+import { AzureDevOpsFieldName } from "./azureDevOpsFieldName";
+import { AzureDevOpsWorkItemType } from "./azureDevOpsWorkItemType";
+import { AzDOJsonPatchDocument, AzDoOp } from "./jsonPatchDocument";
+
+interface FieldValuePair {
+  field: AzureDevOpsFieldName;
+  value?: string;
+}
+
+export class AzureDevOpsUtils {
+  public static createPatchDocument(backlogItem: BacklogItem, parentUrl?: string): AzDOJsonPatchDocument {
+    const pairs = this.createFieldValuePairs(backlogItem);
+    
+    // Only include pairs whose value is defined
+    const document: AzDOJsonPatchDocument = pairs
+      .filter(pair => pair.value)
+      .map((pair: FieldValuePair) => {
+        return {
+          op: AzDoOp.Add,
+          path: `/fields/${pair.field}`,
+          value: pair.value as string,
+        };
+      });
+    
+    if (parentUrl) {
+      document.push({
+        op: AzDoOp.Add,
+        path: "/relations/-",
+        value: {
+          rel: "System.LinkTypes.Hierarchy-Reverse",
+          url: parentUrl,
+          attributes: {
+            isLocked: false,
+            name: "Parent",
+          }
+        }
+      });
+    }
+    
+    return document;
+  }
+
+  public static getWorkItemType(backlogItemType: BacklogItemType): AzureDevOpsWorkItemType {
+    switch (backlogItemType) {
+      case BacklogItemType.Task:
+        return AzureDevOpsWorkItemType.Task;
+      case BacklogItemType.Story:
+        return AzureDevOpsWorkItemType.UserStory;
+      case BacklogItemType.Feature:
+        return AzureDevOpsWorkItemType.Feature;
+      case BacklogItemType.Epic:
+        return AzureDevOpsWorkItemType.Epic;
+      case BacklogItemType.Bug:
+        return AzureDevOpsWorkItemType.Bug;
+      default:
+        throw new Error(`Unsupported backlog item type: ${backlogItemType}`);
+    }
+  }
+
+  public static getBacklogItemType(workItemType: AzureDevOpsWorkItemType): BacklogItemType {
```

Comment:
This is smelly. Can we use inheritence here somehow?

Reply:
Not entirely sure how... Since these are just enums with slightly different string values, we just need some way to map between them